File: examples/cpp/my-app/src/utils/Logger.cpp
#include "utils/Logger.h"
#include "utils/DateUtils.h"
#include <iostream>

namespace utils {

Logger::Logger() {
    logFile.open("app.log", std::ios::app);
}

Logger::~Logger() {
    if (logFile.is_open()) {
        logFile.close();
    }
}

Logger& Logger::getInstance() {
    static Logger instance;
    return instance;
}

void Logger::info(const std::string& message) {
    std::string timestamp = DateUtils::formatDate(DateUtils::now());
    std::string logMessage = "[INFO] " + timestamp + " - " + message;
    std::cout << logMessage << std::endl;
    if (logFile.is_open()) {
        logFile << logMessage << std::endl;
    }
}

void Logger::warning(const std::string& message) {
    std::string timestamp = DateUtils::formatDate(DateUtils::now());
    std::string logMessage = "[WARNING] " + timestamp + " - " + message;
    std::cout << logMessage << std::endl;
    if (logFile.is_open()) {
        logFile << logMessage << std::endl;
    }
}

void Logger::error(const std::string& message) {
    std::string timestamp = DateUtils::formatDate(DateUtils::now());
    std::string logMessage = "[ERROR] " + timestamp + " - " + message;
    std::cerr << logMessage << std::endl;
    if (logFile.is_open()) {
        logFile << logMessage << std::endl;
    }
}

void Logger::debug(const std::string& message) {
    std::string timestamp = DateUtils::formatDate(DateUtils::now());
    std::string logMessage = "[DEBUG] " + timestamp + " - " + message;
    std::cout << logMessage << std::endl;
    if (logFile.is_open()) {
        logFile << logMessage << std::endl;
    }
}

} // namespace utils

File: examples/python/my-app/src/utils/logger.py
"""Logging utility with singleton pattern."""

import logging
from datetime import datetime
from typing import Optional

class Logger:
    """Singleton logger for application-wide logging."""
    
    _instance: Optional["Logger"] = None
    _logger: Optional[logging.Logger] = None
    
    def __init__(self):
        if Logger._instance is not None:
            raise Exception("Logger is a singleton. Use get_instance()")
        
        self._logger = logging.getLogger("TaskApp")
        self._logger.setLevel(logging.INFO)
        
        # Console handler
        console_handler = logging.StreamHandler()
        console_handler.setLevel(logging.INFO)
        
        # Formatter
        formatter = logging.Formatter(
            "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
        )
        console_handler.setFormatter(formatter)
        
        self._logger.addHandler(console_handler)
    
    @staticmethod
    def get_instance() -> "Logger":
        """Get the singleton instance of Logger."""
        if Logger._instance is None:
            Logger._instance = Logger()
        return Logger._instance
    
    def info(self, message: str) -> None:
        """Log info message."""
        self._logger.info(message)
    
    def error(self, message: str) -> None:
        """Log error message."""
        self._logger.error(message)
    
    def warning(self, message: str) -> None:
        """Log warning message."""
        self._logger.warning(message)
    
    def debug(self, message: str) -> None:
        """Log debug message."""
        self._logger.debug(message)
